### ISSUE TITLE
No error for missing argument in type comment if function nested within method

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#5806]
There is no error reported for `g` even though the type comment is clearly missing an argument type:

```py
class A:
    def f(self): 
        # type: () -> None
        def g(x): 
            # type: () -> None  # No error
            pass

def f2(): 
    # type: () -> None
    def g2(x): 
        # type: () -> None  # Correctly reported
        pass
```
